ts/ts/main.ts

```diff
@@ -585,3 +696,148 @@ function favoriteDate(): void {
     saveDate();
   }
 }
+
+if (!$openButton) throw new Error('$openButton not found!');
+$openButton.addEventListener('click', openHolidays);
+
+function toggleSavedHolidayPlaceholder(): void {
+  if (!$savedHolidayPlaceholder)
+    throw new Error('$savedHolidayPlaceholder not found!');
+  if (!$openDialogContent) throw new Error('$openDialogContent not found!');
+  $openDialogContent.children.length > 0
+    ? $savedHolidayPlaceholder.classList.add('hidden')
+    : $savedHolidayPlaceholder.classList.remove('hidden');
+}
+
+function fillOpenDialog(savedHolidaysArray: SavedHoliday[]): void {
+  if (!$openDialogContent) throw new Error('$openDialogContent not found!');
+  toggleSavedHolidayPlaceholder();
+  for (let i = 0; i < savedHolidaysArray.length; i++) {
+    const container = document.createElement('div');
+    container.className = `relative flex flex-col flex-wrap gap-2 w-64
```

Comment:
Would want to revisit this possibly in the future. The function loops through and creates the saved holidays in the dialog. Ideally we make this a `renderSavedHoliday` function. That way, when a holiday is saved you can render the currently saved holiday then and there and then you could loop through the saved holidays on load and run the render function on them.